autogpt_platform/backend/backend/server/routers/integrations.py

```diff
@@ -80,20 +81,63 @@ async def callback(
     user_id: Annotated[str, Depends(get_user_id)],
     request: Request,
 ) -> CredentialsMetaResponse:
-    handler = _get_provider_oauth_handler(request, provider)
+    logger.debug(f"Received callback for provider: {provider}")
+    try:
+        handler = _get_provider_oauth_handler(request, provider)
+    except Exception as e:
+        logger.error(f"Error getting OAuth handler for provider {provider}: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Internal server error: {str(e)}")
 
     # Verify the state token
-    if not await store.verify_state_token(user_id, state_token, provider):
-        raise HTTPException(status_code=400, detail="Invalid or expired state token")
+    try:
+        if not await store.verify_state_token(user_id, state_token, provider):
+            logger.warning(f"Invalid or expired state token for user {user_id}")
+            raise HTTPException(
+                status_code=400, detail="Invalid or expired state token"
+            )
+    except Exception as e:
+        logger.error(f"Error verifying state token: {str(e)}")
+        raise HTTPException(
+            status_code=500, detail=f"Error verifying state token: {str(e)}"
+        )
```

Comment:
Wrapping everything in try/except like this doesn't help: if the token is invalid, it raises an HTTP 400 which is then turned into an HTTP 500 (wrong code for that failure mode) by the `except`.

Reply:
you're correct. fixed